drivers/flash/flash_mcux_flexspi_nor.c

```diff
@@ -886,6 +908,52 @@ static int flash_flexspi_nor_check_jedec(struct flash_flexspi_nor_data *data,
 
 	/* Switch on manufacturer and vendor ID */
 	switch (vendor_id & 0xFFFF) {
+	case 0x609d: /* IS25LP flash, needs P[4:3] cleared with same method as IS25WP */
+		read_params = 0xE0U;
```

Comment:
Add `__fallthrough`? But the `read_params` is reset on line 919.

Reply:
I don't know how I missed this- thanks for catching it. I guess the IS25LP flash will function correctly with P[7:0] set to 0x0, but it is not correct for what we are trying to change here. Updated this section to use a helper function for both flashes.